File: Insertion_Sort/insertion_sort.py
def insertion_sort(l):
    """
    A sorting algorithm that places an unsorted element at its suitable place in each iteration.
    """
    for i in range(len(l) - 1):
        j = i
        while l[j] > l[j + 1]:
            l[j], l[j + 1] = l[j + 1], l[j]
            j = max(j - 1, 0)

    return l
     
        
# Driver code to test above
lst = [6, 1, 5, 9, 9, 5, 3, 8, 0]
sorted_list = insertion_sort(lst)
print(sorted_list)

File: Insertion_Sort/insertion_sort.js
const insertionSort = (arr) => {
    for(let i = 0; i < arr.length - 1; i++) {
        let j = i;
        while(j >= 0 && arr[j] > arr[j+1]) {
            let temp = arr[j];
            arr[j] = arr[j+1];
            arr[j+1] = temp;  
            j--;
        }
    }
}

// Driver code to test above
let lst = [6, 1, 5, 9, 9, 5, 3, 8, 0]
insertionSort(lst)
console.log(lst)